google-cloud-bigtable/scripts/update-deps.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# This script pulls in upstream updates from the veneer code generator and rebases your local
+# changes on top.
+#
+# Specifically it:
+# - fetches changes from googleapis for updated protos
+# - fetches changes from toolkit for updates to the code generator
+# - fetches api-client-staging to use as a staging area for the generated grpc stubs
+#
+# This is meant to be used conjunction with update-deps.sh to regenerate the underlying client with
+# the latest upstream changes.
+
+set -eE # same as: `set -o errexit -o errtrace`
+handle_error() {
+  echo "Failed, exiting. See deps/output.log for more details"
+  exit 1
+}
+trap 'handle_error' ERR
+
+PROJECT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+WORKDIR="$PROJECT_DIR/deps"
+LOG="$WORKDIR/output.log"
+
+DIRS=( googleapis toolkit api-client-staging )
+
+DOCKER_IMAGE=googleapis/artman:stable
+
+function run {
+  init
+  update
+}
+
+# Make sure all of the dependencies are checked out
+function init() {
+  mkdir -p "$WORKDIR"
+  echo "" > $WORKDIR/output.log
+
+  log "Pulling docker image"
+  docker pull "$DOCKER_IMAGE" >>$LOG 2>&1
+
+  if [ ! -d "$WORKDIR/googleapis" ]; then
+    log "Fetching googleapis"
+    git clone "https://github.com/googleapis/googleapis.git" "$WORKDIR/googleapis" >>$LOG 2>&1
+  fi
+
+  if [ ! -d "$WORKDIR/toolkit" ]; then
+    log "Fetching toolkit"
+    git clone "https://github.com/googleapis/toolkit.git" "$WORKDIR/toolkit" >>$LOG 2>&1
+  fi
+
+  if [ ! -d "$WORKDIR/api-client-staging" ]; then
+    log "Fetching api-client-staging"
+    git clone "https://github.com/googleapis/api-client-staging.git" "$WORKDIR/api-client-staging" >>$LOG 2>&1
+  fi
+}
+
+function update() {
+  for DIR in "${DIRS[@]}"; do
+    log "Updating $DIR"
+    DIR="$WORKDIR/$DIR"
+
+    log "Fetching upstream"
+    git -C "$DIR" fetch origin >>$LOG 2>&1
+
+    log "Dropping uncommitted local changes"
+    git -C "$DIR" reset --hard HEAD >>$LOG 2>&1
```

Comment:
This feels like a grenade. Could we have the script fail if the user has any uncommitted local changes?

Reply:
I originally wrote it that way, but in practice it became an annoyance and I changed it to this. Most of the time, the repos are dirty because they contain generated code from the changes you are testing.

Reply:
how hard would it be to have it automatically clear out auto-generated edits but not manual edits? (You have the lst files...)

Reply:
That should work. But would prefer to delay that as a feature add later. Can we merge this and open an issue for me to generalize the scripts and add safer regen guards?

Reply:
Sure.